content/articles/react_query-data-fetching-and-server-state-management/index.md

```diff
@@ -0,0 +1,207 @@
+[React](https://reactjs.org/) is a powerful client-side JavaScript library. One of the complicated aspects to handle is state management. This pattern gets even more convoluted when dealing with server state. Server state is a bit different as it is asynchronous and persisted remotely. This means that we have to think about updating, caching, or re-fetching data to efficiently manage the state in our React applications. [React Query](https://react-query.tanstack.com/) is a pre-configured library that aims to solve this complexities. Using React Query, we can fetch, cache, and update data in React-based applications in a declarative and simple manner without touching any global state.
+
+### Goal
+Nowadays, almost every web app needs a remote data. Unfortunately, for developers, data fetching and handling server state in React applications is easier said than done. As developers we need to rethink about:
+
+- What to render while waiting for the remote data to load?
+
+- What happens if an error occurs?
+
+- How do we keep the client data upto date with the server?
+
+- How will this tradeoffs impact users on a slow internet?
+
+When dealing with asynchronous data that needs frequent update, caching, and synchronization with the server, there is no better library than React-Query. In this tutorial, we will go through how React-Query library can improve user experience in our react applications. To demonstrate these concepts, we will use the [JSON Placeholder](https://jsonplaceholder.typicode.com/) as the thrid party REST API.
+
+### Prerequisites
+- For this tutorial, intermediate React kills such as React Hooks and functional components is essential.
+
+- A basic understanding of REST APIs and data fetching in JavaScript.
+  
+- Make sure to have [Node.js](https://nodejs.org/en/) runtime installed on your machine.
+
```

Comment:
Let's have sub-topics talking about 
- State management
- Server state (like you mentioned in the intro)

Reply:
Yes, this is what I'm working on.